cmd/telegraf/telegraf.go

```diff
@@ -111,6 +116,45 @@ func reloadLoop(
 	}
 }
 
+// loadExternalPlugins loads external plugins from shared libraries (.so, .dll, etc.)
+// in the specified directory.
+func loadExternalPlugins(rootDir string) error {
+	return filepath.Walk(rootDir, func(pth string, info os.FileInfo, err error) error {
+		// Stop if there was an error.
+		if err != nil {
+			return err
+		}
+
+		// Ignore directories.
+		if info.IsDir() {
+			return nil
+		}
+
+		// Ignore files that aren't shared libraries.
+		ext := strings.ToLower(path.Ext(pth))
+		if ext != ".so" && ext != ".dll" {
+			return nil
+		}
+
+		// name will be the path to the plugin file beginning at the root
+		// directory, minus the extension.
+		// ie, if the plugin file is ./telegraf-plugins/foo.so, name
+		// will be "telegraf-plugins/foo"
+		name := strings.TrimPrefix(strings.TrimPrefix(pth, rootDir), string(os.PathSeparator))
+		name = strings.TrimSuffix(name, filepath.Ext(pth))
```

Comment:
Why is the var ``name`` defined?
Looks like is not being used.

Maybe it was thought to register the plugin with that name?

Reply:
Also, TOML doesn't like names with slasesh:
```
[[inputs.extplugins/ext-kernel]]
```

```
2019-08-05T11:01:53Z E! [telegraf] Error running agent: Error parsing prueba.conf, line 5: invalid TOML 
```

Reply:
The `name` is leftover of the previous implementation, you're right that it should be removed.

> Also, TOML doesn't like names with slasesh

You can include a slash using ""

```
[[inputs."extplugins/ext-kernel"]]
```

Reply:
You probably already have this figured out, but I removed the name code when merging. 
Plugins can be loaded using the name specified in the registration function called from init().

I also have an example that I was using for testing over at https://github.com/danielnelson/telegraf-plugins